hw2-turtle-graphics/turtle-stretch2.js

```diff
@@ -0,0 +1,184 @@
+                this.direction = "west"
+
+            } else if (this.direction == "east") {
+                this.direction = "north"
+
+            } else if (this.direction == "south") {
+                this.direction = "east"
+
+            } else if (this.direction == "west") {
+                this.direction = "south"
+            }
+            return this;
+        }
+
+        allPoints(){
+        return(this.allpoints)
+        } 
+        
+        print(){
+        
+            const final = [];
+            const slicedpoints = this.allpoints
+            
+            for (let i=0 ; i<this.maxX+2; i++){         //iterates through the widtn of the grid            
+                const x = []
+                for(let j=0; j<this.maxY+2; j++){  //iterates through the length of the grid
+                    const gridPath = slicedpoints.find((item)=>{
+                    return item.join(",")===`${j},${i}` 
+                    })
+                    if(gridPath) x.push("■") // if the box is crossed by the turtle-- fills the box, pushes to x array
+                    else x.push("□") // if not, the box will be empty- push to x array
+                }
+                
+                final.push(x.join(""))
+                }
+                //changed this to return from the last assignment-- because I don't want it to print to the console, 
+                //I want it to print to a new file so this will take place in the savetofile() function
+                return final.join("\n")
+        }  
+ }
+
+ function stretch() {
+   // taking the arguments, splitting them into an array separated by the dashes
+
+    let commands = process.argv.slice(3);
+    commands = commands.toString()
+    let commandsArr = commands.split("-")
+
+
+    //this iterate through and convert each shorthand command to a function call, eg.f10 will be .forward(10)
+    let turtle = new Turtle(0,0); //setting a default value
+
+    for (let i = 0; i < commandsArr.length; i++) {
+        if (commandsArr[i][0]== "f"){
+            turtle = turtle.forward(commandsArr[i].slice(1))
+        } else if (commandsArr[i][0]== "r") {
+            turtle = turtle.right()
+        } else if (commandsArr[i][0]== "l") {
+            turtle = turtle.left()
+        } else if (commandsArr[i][0]== "t"){
+            turtle = new Turtle(commandsArr[i][1],commandsArr[i][3])
+        } else {
+            console.log("Whoops, try again. Enter something like 't0,4-f3-l-f3-r-f5-r-f8-r-f5-r-f3-l-f3'")
+        } 
+    } 
+        return turtle;
+    }
+//this seemed to fit better outside of a function. Just grabbing the input, and separating the filename from the commands.
+// if --output is included-- call saveToFile function
+
+let input = process.argv[2];
+input = input.toString()
+let inputsplit = input.split("=")
+
+if (inputsplit =="--output"){
+    saveToFile()
+    }
+
+function saveToFile() {
+
+let data = stretch().print();
+let filename = inputsplit[1];
```

Comment:
`filename` should be passed in as an argument to this function rather than grabbing a variable from outside this function - not suggested. So what you should do is
- line 158: `saveToFile(inputSplit[1])`
- line 161: function saveToFile(filename)
- line 164: remove this line